classes/01class/exercises/c01-aws01/gpavelar/README.md

```diff
@@ -0,0 +1,436 @@
+          "EnaSupport": true,
+          "Hypervisor": "xen",
+          "NetworkInterfaces": [
+            {
+              "Attachment": {
+                "AttachTime": "2020-06-19T09:29:05+00:00",
+                "AttachmentId": "eni-attach-0ababbd8be8576030",
+                "DeleteOnTermination": true,
+                "DeviceIndex": 0,
+                "Status": "attached"
+              },
+              "Description": "",
+              "Groups": [
+                {
+                  "GroupName": "devops-develop",
+                  "GroupId": "sg-076a9212c8d1a8a45"
+                }
+              ],
+              "Ipv6Addresses": [],
+              "MacAddress": "02:4b:02:ca:6b:90",
+              "NetworkInterfaceId": "eni-0896f569c813a7cf3",
+              "OwnerId": "478433196210",
+              "PrivateDnsName": "ip-172-31-2-153.ap-southeast-2.compute.internal",
+              "PrivateIpAddress": "172.31.2.153",
+              "PrivateIpAddresses": [
+                {
+                  "Primary": true,
+                  "PrivateDnsName": "ip-172-31-2-153.ap-southeast-2.compute.internal",
+                  "PrivateIpAddress": "172.31.2.153"
+                }
+              ],
+              "SourceDestCheck": true,
+              "Status": "in-use",
+              "SubnetId": "subnet-351ced53",
+              "VpcId": "vpc-37d4dd50",
+              "InterfaceType": "interface"
+            }
+          ],
+          "RootDeviceName": "/dev/xvda",
+          "RootDeviceType": "ebs",
+          "SecurityGroups": [
+            {
+              "GroupName": "devops-develop",
+              "GroupId": "sg-076a9212c8d1a8a45"
+            }
+          ],
+          "SourceDestCheck": true,
+          "VirtualizationType": "hvm",
+          "CpuOptions": {
+            "CoreCount": 1,
+            "ThreadsPerCore": 1
+          },
+          "CapacityReservationSpecification": {
+            "CapacityReservationPreference": "open"
+          },
+          "HibernationOptions": {
+            "Configured": false
+          },
+          "MetadataOptions": {
+            "State": "applied",
+            "HttpTokens": "optional",
+            "HttpPutResponseHopLimit": 1,
+            "HttpEndpoint": "enabled"
+          }
+        }
+      ],
+      "OwnerId": "478433196210",
+      "ReservationId": "r-0927dd5e49e7f6d40"
+    }
+  ]
+}
+```
+
+```bash
+## Private IP - 172.31.3.15
+ssh -i /var/scratch/DevOps_Academy/DevOps-academy.pem  ec2-user@172.31.2.153
+ssh: connect to host 172.31.2.153 port 22: Connection timed out
+```
+
+- It's not possible to connect via SSH, because it is not available to be accessed.
+
+So we will overcome this step using [SSH Agent](https://aws.amazon.com/blogs/security/securely-connect-to-linux-instances-running-in-a-private-amazon-vpc/). But we need to adapt the command because the OS ([ubuntu command here](https://blog.earth-works.com/2015/01/29/add-an-ec2-pem-key-to-ssh-on-linux-and-mac-os-x/))
```

Comment:
hum, did the key exist in your public instance?

you are using the key `devops-class` for both of them, so either you copy the text of the key from your host (laptop) to the public instance and recreate the .pem file there, or you can using the key forwarder via ssh-agent (as you did)

Reply:
I was able to connect to the public instance using ssh -A. But I couldn't connect directly to the private ( that is correct), so I connected first to the public and after that i connected to the private instance. I copied the wrong ssh attempt to the answer, that IP should be the private ec2 instance. I've already updated the answer.